src/datatypes.c

```diff
@@ -6,13 +6,19 @@
 #include <stdlib.h>
 #include <string.h>
 
-edge_t* make_edges(char* all_edges, size_t rotation) {
-  char rotated[4];
-  memcpy(rotated + rotation, all_edges, 4 - rotation);
-  memcpy(rotated, all_edges + 4 - rotation, rotation);
-  edge_t edges = {rotated};
-  return &edges;
-}
+edge_t* make_edges(char* all_edges) {
+  char *temp = malloc(sizeof(char)*4);
+  temp = all_edges;
+  edge_t *edges= malloc(sizeof(edge_t));
+  edges->tile_edges = temp;
+  return edges;
+}// this needs a sanity check
+
+void rotate_edges(edge_t *edges, size_t rotation) {
+  char *temp = edges->tile_edges;
```

Comment:
(moving comment) I think this makes a shallow copy so changing `edges->tile_edges` also changes temp, defeating the purpose

Reply:
I think to fix this we should malloc a temp, strncpy/memcpy `edges->tile_edges` into temp, then free temp at the end